includes/class-wc-siftscience-admin.php

```diff
@@ -253,86 +249,69 @@ public function add_settings_page( $pages ) {
-				$this->get_text_input( WC_SiftScience_Options::API_KEY,
+				$this->get_element( 'text', WC_SiftScience_Options::API_KEY,
 					'Rest API Key', 'The API key for production' ),
 
-				$this->get_text_input( WC_SiftScience_Options::JS_KEY,
+				$this->get_element( 'text', WC_SiftScience_Options::JS_KEY,
 					'Javascript Snippet Key', 'Javascript snippet key for production' ),
 
-				$this->get_number_input( WC_SiftScience_Options::THRESHOLD_GOOD,
-					'Good Score Threshold', 'Scores below this value are considered good and shown in green', 30),
+				$this->get_element( 'number', WC_SiftScience_Options::THRESHOLD_GOOD,
+					'Good Score Threshold', 'Scores below this value are considered good and shown in green', array( 'default' => 30) ),
 
-				$this->get_number_input( WC_SiftScience_Options::THRESHOLD_BAD,
-					'Bad Score Threshold', 'Scores above this value are considered bad and shown in red', 60 ),
+				$this->get_element( 'number', WC_SiftScience_Options::THRESHOLD_BAD,
+					'Bad Score Threshold', 'Scores above this value are considered bad and shown in red', array( 'default' => 60 ) ),
 
-				$this->get_text_input( WC_SiftScience_Options::NAME_PREFIX,
+				$this->get_element( 'text', WC_SiftScience_Options::NAME_PREFIX,
 					'User & Order Name Prefix',
 					'Prefix to give order and user names. '
 					. 'Useful when you have have multiple stores and one Sift account.' ),
 
-				$this->get_check_box( WC_SiftScience_Options::AUTO_SEND_ENABLED,
+				$this->get_element( 'checkbox', WC_SiftScience_Options::AUTO_SEND_ENABLED,
 					'Automatically Send Data',
 					'Automatically send data to Sift when an order is created'
 				),
 
-				$this->get_number_input( WC_SiftScience_Options::MIN_ORDER_VALUE,
+				$this->get_element( 'number', WC_SiftScience_Options::MIN_ORDER_VALUE,
 					'Minimum Order Value for Auto Send',
-					'Orders less than this value will not be automatically sent to sift. Set to zero to send all orders.', 0 ),
+					'Orders less than this value will not be automatically sent to sift. Set to zero to send all orders.', array( 'default' => 0 ) ),
 
-				$this->get_section_end( 'sifsci_section_main' ),
+				$this->get_element( 'sectionend','sifsci_section_main' ),
 			);
 		}
 
-		private function get_title( $id, $title ) {
-			return array( 'title' => $title, 'type' => 'title', 'desc' => '', 'id' => $id );
-		}
-
-		private function get_text_input( $id, $title, $desc ) {
-			return array(
-				'title' => $title,
-				'desc' => $desc,
-				'desc_tip' => true,
-				'type' => 'text',
-				'id' => $id,
-			);
-		}
+		private function get_element( $type, $id, $title = '', $desc = '', $special_options = array() ) {
+			
+			$element = array( 'type' => $type, 'id' => $id );
+
+			switch ( $type ) {
+				case 'sectionend':
+					return $element;
+				case 'title':
+					return array_merge( $element, array( 'title' => $title, 'desc' => $desc ) );
+				case 'number':
+				case 'select':
+					if ( ! empty( $special_options ) ) {
```

Comment:
if `type` is "select", I think $options ($special_options) cannot and should not be empty.

You should either assume it's defined, or throw an error if it's not.

Reply:
future add-ons may be dynamic

Reply:
But a `select` with no options has no meaning.